### DISCUSSION TITLE
Application error: a client-side exception has occurred (see the browser console for more information).

### DISCUSSION THREAD

**Discussion by zpenct** [vercel/next.js#48775] [App Router]
You can check my web here for test experience : https://quick-set-app.vercel.app/
You can check my code,  sorry because it still messy : https://github.com/fayzulhq/setupApp

are my routes error? my page is empty even though I have created page.js for the route

So, the probelm is in my routing on header/navigation....So i have 3 button,:
1) Home  : .../
2) Documentations : .../documentations
it can be detail  :  .../documentations/react
3) Others :  .../others 
but it is not complete yet

So, i am in home `/` and than i click Documentations button, it will redirect me to  .../documentations .So i am in `/documentations`, if i click Home button the error is showed up.

this is my project root

<img width="274" alt="Screen Shot 2023-04-24 at 21 21 26" src="https://user-images.githubusercontent.com/118333594/234008919-a46d8ef8-6470-4631-9e1d-ed7759a950d1.png">

//route
```javascript
import CardItem from '../components/CardItem'

const data = [
  {
    title : 'Install NPM',
    link : '/documentations/npm'
  },
  {
    title : 'Setup React JS',
    link : '/documentations/react'
  },
  {
    title : 'Setup React JS + Tailwind + CRA',
    link : '/documentations/react-tailwind-cra'
  },
  {
    title : 'Setup React JS + Tailwind + Vite',
    link : '/documentations/react-tailwind-vite'
  },
  {
    title : 'Setup Next.Js 13 + Tailwind',
    link : '/documentations/next-tailwind'
  },
]

const CardContainer = () => {
  return (
    <div className='flex flex-col justify-center px-10'>
      {data.map(({title,link})=>(
        <CardItem title={title} link={link} key={title}/>
      ))}
    </div>
  )
}
``` `
//Header
```javascript
const Header = (props) => {
  
  return (
    <div className="m-4">
      <div className="flex justify-between p-3 md:p-5 max-w-4xl mx-auto items-center">
        <Link href='/' className={`${style} hidden md:flex`}>
          Quick
        </Link>
        <div className="flex justify-center mx-auto md:mx-0 gap-x-7">
          <Link href="/documentations" className={style}>
            Documentations
          </Link>
          <Link href="/others" className={style}>
            Others
          </Link>
        </div>
      </div>
    </div>
  );
};

// export default Header;

export default dynamic (()=> Promise.resolve(Header),{ssr:false})
```

**Comment by icyJoseph** [comment]
Hi,

I think the problem is that your `documentations/` route, is defining a nested layout, which applies `html`, `head`, and `body`, so you create a nested `html` page, within an `html` page.

I'd recommend reading about layouts first: https://beta.nextjs.org/docs/routing/pages-and-layouts

Your layout in `documentations` should only have to be like:

```jsx
import "./../globals.css";
import Header from "./../components/header";
import { ContactUs } from "../components/Contact";

export const metadata = {
  title: "Tech",
};
export default function Layout({ children }) {
  return (
    <div className="max-w-3xl mx-auto my-24">{children}</div>
  );
}
```

If, any at all, I haven't paid too much attention to what you are actually, trying to accomplish, just fixing the error. 

### next/dynamic

Next dynamic is to break bundles into separate chunks, so defining a dynamic import, within the module, is an anti-pattern. `next/dynamic` should receive the `import` function, with a predictable path:

```js
// in the header have this
export default Header;
```

And then:

```js
const Header = dynamic(() => import("./components/header", { ssr: false }));
```

> Note: In import('path/to/component'), the path must be explicitly written. It can't be a template string nor a variable. Furthermore the import() has to be inside the dynamic() call for Next.js to be able to match webpack bundles / module ids to the specific dynamic() call and preload them before rendering. dynamic() can't be used inside of React rendering as it needs to be marked in the top level of the module for preloading to work, similar to React.lazy.

[Source](https://nextjs.org/docs/advanced-features/dynamic-import#example)

**Reply by zpenct** [reply]
I still don't understand dynamic nextJs, what should I write in my code if my overall code is like this...sorry i am newbie...Please don’t be mad at me.

```javascript
import Link from "next/link";
import dynamic from "next/dynamic";

const style = 'text-sm ...'

const Header = (props) => {
  
  return (
    <div className="m-4">
      <div className="flex justify-between p-3 md:p-5 max-w-4xl mx-auto items-center">
        <Link href='/' className={`${style} hidden md:flex`}>
          Quick
        </Link>
        <div className="flex justify-center mx-auto md:mx-0 gap-x-7">
          <Link href="/documentations" className={style}>
            Documentations
          </Link>
          <Link href="/others" className={style}>
            Others
          </Link>
        </div>
      </div>
    </div>
  );
};

// export default Header;

export default dynamic (()=> Promise.resolve(Header),{ssr:false})
```  
where do i write this code? in Layout.js
```const Header = dynamic(() => import("./components/header", { ssr: false }));```

**Reply by zpenct** [reply]
thanks sir, i works...thanks

**Reply by icyJoseph** [reply]
Don't forget to mark as answered :) and no worries, happy to help!

You write the dynamic import where you use the component, so that it can be downloaded as its own JS chunk.

In this case, this would go on `layout.js`:

```js
// some imports and other stuff
const Header = dynamic(() => import("./components/header", { ssr: false }));
// rest of layout
```

**Comment by vvda12061999** [comment]
I solve this by running

`npx create-next-app@latest`

Then I move all of the /src files in the bugged project to a new one (including all .git files), then I fix some errors, and it works normally.

Then I go for `yarn install` and `yarn dev`
